test/vstest.console.UnitTests/TestPlatformHelpers/TestRequestManagerTests.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace vstest.console.UnitTests.TestPlatformHelpers
+{
+    using System.Collections.Generic;
+    using Microsoft.VisualStudio.TestPlatform.Client.RequestHelper;
+    using Microsoft.VisualStudio.TestPlatform.CommandLine;
+    using Microsoft.VisualStudio.TestPlatform.CommandLine.TestPlatformHelpers;
+    using Microsoft.VisualStudio.TestPlatform.Common.Interfaces;
+    using Microsoft.VisualStudio.TestPlatform.Common.Logging;
+    using Microsoft.VisualStudio.TestPlatform.CoreUtilities.Tracing.Interfaces;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client.Interfaces;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Moq;
+    using vstest.console.UnitTests.TestDoubles;
+
+    [TestClass]
+    public class TestRequestManagerTests
+    {
+        private DummyLoggerEvents mockLoggerEvents;
+        private TestLoggerManager mockLoggerManager;
+        private CommandLineOptions commandLineOptions;
+        private Mock<ITestPlatform> mockTestPlatform;
+        private TestRequestManager testRequestManager;
+        private Mock<IDiscoveryRequest> mockDiscoveryRequest;
+        private Mock<ITestRunRequest> mockRunRequest;
+
+        public TestRequestManagerTests()
+        {
```

Comment:
MSTest always calls ctor for every TestMethod. The mvp bug bash issue is a bug :)